<?xml version="1.0" encoding="UTF-8"?>
<robot name="robot" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="0.6" radius="0.2"/>
      </geometry>

    </visual>
    <collision>
      <geometry>
        <cylinder length="0.6" radius="0.2"/>
      </geometry>
    </collision>
  </link>
  <link name="test">
    <visual>
      <geometry>
        <cylinder length="0.6" radius="0.2"/>
      </geometry>

    </visual>
    <collision>
      <geometry>
        <cylinder length="0.6" radius="0.2"/>
      </geometry>
    </collision>
  </link>
  <joint name="vesc_joint" type="continuous">
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <parent link="base_link"/>
    <child link="test"/>
  </joint>
  <ros2_control name="vesc" type="actuator">
    <hardware>
      <plugin>vesc_hw_interface/VescHwInterface</plugin>
      <param name="port">/dev/ttyACM0</param>
      <param name="gear_ratio">0.230769</param>
      <param name="torque_const">1.0</param>
      <param name="num_hall_sensors">3</param>
      <param name="screw_lead">1.0</param>
      <param name="num_rotor_poles">20</param>
      <param name="motor/Kp">0.1</param>
      <param name="motor/Ki">0.01</param>
      <param name="motor/Kd">0.02</param>
      <param name="motor/i_clamp">1.0</param>
      <param name="motor/duty_limiter">1.0</param>
      <param name="motor/antiwindup">true</param>
      <param name="motor/control_rate">100.0</param>
      <param name="motor/enable_smooth_diff">true</param>
      <param name="motor/smooth_diff/max_sample_sec">0.2</param>
      <param name="motor/smooth_diff/max_smooth_step">10</param>
      <param name="command_mode">velocity_duty</param>
      <param name="joint_type">continuous</param>
    </hardware>
    <joint name="vesc_joint">
      <command_interface name="position">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="velocity">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <command_interface name="effort">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
    </joint>
  </ros2_control>
</robot>
